content/blog/2022-04-04-develop-and-test-data-analytics-pipelines-on-localstack/index.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Develop and Test Real-time Data Pipelines with LocalStack
+description: "LocalStack makes it easy to develop and test real-time data pipelines that are built with AWS services. Learn how to deploy Cloudwatch, Kinesis, Lambda, and external services using CDK on LocalStack."
+lead: "LocalStack makes it easy to develop and test real-time data pipelines that are built with AWS services. Learn how to deploy Cloudwatch, Kinesis, Lambda, and external services using CDK on LocalStack."
+date: 2022-04-04
+lastmod: 2022-04-04
+draft: false
+images: []
+contributors: ["Sam Watson"]
+tags: ["tutorial"]
+---
+
+{{< img src="teaser.png" >}}
+
+At LocalStack, we rely on AWS Lambda as a key part of our serverless infrastructure toolkit. As with any critical service, we want to extract analytics event data from running Lambda functions and aggregate it in our data warehouse so that we can track user stories and squash bugs. But this presents a challenge: how can we emit detailed analytics data while keeping Lambda code simple and performant? To solve this, we developed a serverless streaming data pipeline using CloudWatch Logs and Kinesis that allows us to decouple decouple analytics from application logic.
```

Comment:
```suggestion
At LocalStack, we rely on AWS Lambda as a key part of our serverless infrastructure toolkit. As with any critical service, we want to extract analytics event data from running Lambda functions and aggregate it in our data warehouse so that we can track user stories and squash bugs. But this presents a challenge: how can we emit detailed analytics data while keeping Lambda code simple and performant? To solve this, we developed a serverless streaming data pipeline using CloudWatch Logs and Kinesis that allows us to decouple analytics from application logic.
```